### PR TITLE
Plugin testing

### PR DESCRIPTION
**Pull request by michael0525**
This feature branch sets up the WordPress PHPUnit test environment and performs two unit test for the prototype plugin. The first test is to test if the sets of events in the active plugins are included before we want to inject the code for event tracking. The second test is to test if the expected Javascript code is injected for event tracking.

### CODE REVIEW

File: tests/phpunit/MeasurementEventTest.php [hunk 1]
```diff
@@ -64,4 +64,79 @@ public function testEventConfigurations() {
...
+                            addListeners(configCopy);
+                        });
+                    }
+                }
+            });
+        </script>";
+
+        $this->expectOutputString($expectedString);
+    }
 }
```
Comment (yassinh-g) [inline tests/phpunit/MeasurementEventTest.php]: Add new line at end of the file

File: src/MeasurementCodeInjector.php [hunk 2]
```diff
@@ -38,37 +30,44 @@ class MeasurementCodeInjector {
...
+     * Sets the event configurations
      */
-    public function setActivePlugins() {
-        $this->activePlugins = $this->pluginDetector->getActivePlugins();
-    }
-
-    /**
-     * Creates list of measurement event configurations and javascript to inject
-     */
-    public function injectEventTracking() {
```
Comment (yassinh-g) [inline src/MeasurementCodeInjector.php:61]: getEventConfigurations or buildEventConfigurations

File: tests/phpunit/MeasurementEventTest.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+        $this->assertEquals(count($measurementEvents), count($eventConfigurations));
+
+        $found = false;
+
+        for ($i = 0; $i < count($measurementEvents); $i++) {
+            $found = false;
+            foreach($eventConfigurations as $currentEvent) {
+                if ($measurementEvents[$i]->getPluginName() == $currentEvent->getPluginName() &&
+                $measurementEvents[$i]->getSelector() == $currentEvent->getSelector()) {
```
Comment (yassinh-g) [inline tests/phpunit/MeasurementEventTest.php]: json_encode($measurementEvents[$i]) === json_encode($currentEvent)

Would this work?

**Review by yassinh-g — DISMISSED** [review]

File: tests/phpunit/MeasurementEventTest.php (hunk 3 above)
Comment (michael0525) [reply tests/phpunit/MeasurementEventTest.php]: This works. I have changed it to the json_encode() comparison between the two events.

**Review by yassinh-g — APPROVED** [review]

**Pull request merged**